From 011c09046070f1bd35e18c65f42112e40b8fe53c Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Wed, 30 Sep 2015 13:44:14 +0300
Subject: [PATCH 04/26] Subject: [PATCH] Themes: Allow notification shadows to
 be rounded

If a theme has a very large rounded rectangle it is obvious
the notification's shadow is a rectangle. So this patch
set the shadow to be drawn as a rounded rect like the notification itself.

Change-Id: I560fc5c4a4ce3d052d98b14af978a28c13b6caca
---
 .../statusbar/ActivatableNotificationView.java     |  1 +
 .../systemui/statusbar/ExpandableOutlineView.java  | 22 +++++++++++++---------
 2 files changed, 14 insertions(+), 9 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/ActivatableNotificationView.java b/packages/SystemUI/src/com/android/systemui/statusbar/ActivatableNotificationView.java
index 465a141..780062f 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/ActivatableNotificationView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/ActivatableNotificationView.java
@@ -154,6 +154,7 @@ public abstract class ActivatableNotificationView extends ExpandableOutlineView
         mAppearAnimationFilter = new PorterDuffColorFilter(0, PorterDuff.Mode.SRC_ATOP);
         mRoundedRectCornerRadius = getResources().getDimensionPixelSize(
                 R.dimen.notification_material_rounded_rect_radius);
+        setRoundCornerRadius(mRoundedRectCornerRadius); // Themes: For drop-shadow rounded corners
         mLegacyColor = getResources().getColor(R.color.notification_legacy_background_color);
         mNormalColor = getResources().getColor(R.color.notification_material_background_color);
         mLowPriorityColor = getResources().getColor(
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/ExpandableOutlineView.java b/packages/SystemUI/src/com/android/systemui/statusbar/ExpandableOutlineView.java
index a18fff2..0690555 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/ExpandableOutlineView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/ExpandableOutlineView.java
@@ -31,20 +31,21 @@ public abstract class ExpandableOutlineView extends ExpandableView {
 
     private final Rect mOutlineRect = new Rect();
     private boolean mCustomOutline;
+    private float mRoundCornerRadius = 0;
 
     public ExpandableOutlineView(Context context, AttributeSet attrs) {
         super(context, attrs);
         setOutlineProvider(new ViewOutlineProvider() {
             @Override
-            public void getOutline(View view, Outline outline) {
-                if (!mCustomOutline) {
-                    outline.setRect(0,
-                            mClipTopAmount,
-                            getWidth(),
-                            Math.max(getActualHeight(), mClipTopAmount));
-                } else {
-                    outline.setRect(mOutlineRect);
-                }
+             public void getOutline(View view, Outline outline) {
+                 if (!mCustomOutline) {
+                    outline.setRoundRect(0,
+                             mClipTopAmount,
+                             getWidth(),
+                            Math.max(getActualHeight(), mClipTopAmount), mRoundCornerRadius);
+                 } else {
+                    outline.setRoundRect(mOutlineRect, mRoundCornerRadius);
+                 }
             }
         });
     }
@@ -82,4 +83,7 @@ public abstract class ExpandableOutlineView extends ExpandableView {
         invalidateOutline();
     }
 
+    protected void setRoundCornerRadius(float roundRadius) {
+        mRoundCornerRadius = roundRadius;
+    }
 }
-- 
1.9.1

